components/api/src/main/java/com/hotels/styx/api/FullHttpRequest.java

```diff
@@ -55,7 +55,30 @@
 import static java.util.stream.Stream.concat;
 
 /**
- * HTTP request with a fully aggregated/decoded body.
+ * An immutable HTTP request object including full body content.
+ * <p>
+ * A {@link FullHttpRequest} is useful for request messages with a
```

Comment:
It might be better to just say "requests" instead of "request messages", even though they both make sense. I think HTTP requests are typically just referred to that way.

Perhaps we could go with "A {@link FullHttpRequest} is useful for representing requests with ..."

Reply:
It would have to say:

> A {@link FullHttpRequest} represents requests with ...

Reply:
I'd leave it like this:
``` 
 * A {@link FullHttpRequest} is useful for requests with a
 * finite body content, such as when PUT or POST are used to create or
 * modify a RESTful object.
```